### PR TITLE
fix: use 1 for beam search candidate length for the levels above the assigned level

### PR DESCRIPTION
**Pull request by eddyxu**
To be consistent with the paper.

### CODE REVIEW

**Review by chebbyChefNEQ — APPROVED** [review]

File: rust/lance-index/src/vector/hnsw/builder.rs [hunk 1]
```diff
@@ -159,7 +159,7 @@ impl HNSWBuilder {
         //  }
         // ```
         for cur_level in self.levels.iter().rev().take(levels_to_search) {
-            let candidates = beam_search(cur_level, &ep, vector, self.ef_construction)?;
+            let candidates = beam_search(cur_level, &ep, vector, 1)?;
             ep = if self.use_select_heuristic {
                 select_neighbors_heuristic(cur_level, query, &candidates, 1, self.extend_candidates)
                     .map(|(_, id)| id)
```
Comment (BubbleCal) [inline rust/lance-index/src/vector/hnsw/builder.rs:165]:
```suggestion
```
these could be all removed as the `beam_search` returns only one candidate

Comment (eddyxu) [reply rust/lance-index/src/vector/hnsw/builder.rs:165]: Good point, Actually, is `select_heuristic(exstend_candidate=False)` just `select_simple`?

Comment (eddyxu) [reply rust/lance-index/src/vector/hnsw/builder.rs]: Fixed.

Comment (BubbleCal) [reply rust/lance-index/src/vector/hnsw/builder.rs]: > Good point, Actually, is `select_heuristic(exstend_candidate=False)` just `select_simple`?

for the current impl it's the same but likely it doesn't follow the paper

**Review by BubbleCal — APPROVED** [review]

**Pull request merged**